101-iothub-auto-route-messages/azuredeploy.json

```diff
@@ -0,0 +1,172 @@
+            "metadata": {
+                "description": "Name of the endpoint for the storage account."
+            }
+        }
+    },
+    "variables": {
+        "iotHubName": "[concat(parameters('IotHubNamePrefix'), uniqueString(resourceGroup().id))]",
+        "storageAccountName": "[concat(parameters('storageAccountNamePrefix'), uniqueString(resourceGroup().id))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "apiVersion": "2019-06-01",
+            "name": "[variables('storageAccountName')]",
+            "location": "[parameters('location')]",
+            "sku": {
+                "name": "Standard_LRS",
+                "tier": "Standard"
+            },
+            "kind": "Storage",
+            "properties": {},
+            "resources": [
+                {
+                    "type": "blobServices/containers",
+                    "apiVersion": "2019-06-01",
+                    "name": "[concat('default/', parameters('storageContainerName'))]",
+                    "properties": {
+                        "publicAccess": "None"
+                    },
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"
+                    ]
+                }
+            ]
+        },
+        {
+            "type": "Microsoft.Devices/IotHubs",
+            "apiVersion": "2020-03-01",
+            "name": "[variables('IoTHubName')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"
+            ],
+            "properties": {
+                "eventHubEndpoints": {
+                    "events": {
+                        "retentionTimeInDays": 1,
+                        "partitionCount": "[parameters('d2cPartitions')]"
+                    }
+                },
+                "routing": {
+                    "endpoints": {
+                        "storageContainers": [
+                            {
+                                "connectionString": "[Concat('DefaultEndpointsProtocol=https;AccountName=',variables('storageAccountName'),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2019-06-01').keys[0].value)]",
+                                "containerName": "[parameters('storageContainerName')]",
+                                "fileNameFormat": "{iothub}/{partition}/{YYYY}/{MM}/{DD}/{HH}/{mm}",
+                                "batchFrequencyInSeconds": 100,
+                                "maxChunkSizeInBytes": 104857600,
+                                "encoding": "json",
+                                "name": "[parameters('storageEndpoint')]",
+                                "subscriptionId": "[subscription().subscriptionId]",
```

Comment:
is this meant to allow for storage accounts in other subscriptions?  If so, seems like we should account for the scenario - if not, then just remove them.  (or if we don't want to provide the former, just remove them)